src/network/mcpe/InventoryManager.php

```diff
@@ -209,11 +209,11 @@ public function syncSlot(Inventory $inventory, int $slot) : void{
 			$currentItem = $inventory->getItem($slot);
 			$clientSideItem = $this->initiatedSlotChanges[$windowId][$slot] ?? null;
 			if($clientSideItem === null or !$clientSideItem->equalsExact($currentItem)){
-				$typeConverter = TypeConverter::getInstance();
+				$itemStackWrapper = ItemStackWrapper::legacy(TypeConverter::getInstance()->coreItemStackToNet($currentItem));
 				if($windowId === ContainerIds::OFFHAND){
```

Comment:
Please add a `TODO: HACK!` comment about this with a brief explanation why it's needed.